### PR TITLE
refactor: update getListOrderPending and test cases for cancelLimitOrder

### PR DESCRIPTION
**Pull request by DesertEaglePosition**

### CODE REVIEW

File: contracts/protocol/PositionHouse.sol [hunk 1]
```diff
@@ -816,12 +816,13 @@ contract PositionHouse is
         IPositionManager _positionManager,
         address _trader
     ) public view returns (LimitOrderPending[] memory) {
+        address _pmAddress = address(_positionManager);
```
Comment (DesertEaglePosition) [inline contracts/protocol/PositionHouse.sol:819]: assign a memory variable for position manager address to save gas

File: contracts/protocol/libraries/position/PositionHouseFunction.sol [hunk 2]
```diff
@@ -202,7 +202,7 @@ library PositionHouseFunction {
     // TODO edit access modifier cause this function called write function in position manager
     function clearAllFilledOrder(
         IPositionManager _positionManager,
-        PositionLimitOrder.Data[] memory listLimitOrder,
+        PositionLimitOrder.Data[] memory limitOrder,
```
Comment (DesertEaglePosition) [inline contracts/protocol/libraries/position/PositionHouseFunction.sol]: change meaningful variable name

File: contracts/protocol/libraries/position/PositionHouseFunction.sol [hunk 3]
```diff
@@ -378,39 +378,40 @@ library PositionHouseFunction {
...
                             partialFilled: partialFilled,
-                            pip: int256(int128(listLimitOrder[i].pip)),
-                            leverage: listLimitOrder[i].leverage,
-                            blockNumber: listLimitOrder[i].blockNumber,
-                            orderIdOfTrader: i,
-                            orderId: listLimitOrder[i].orderId
+                            pip: limitOrder[i].pip,
+                            leverage: limitOrder[i].leverage,
+                            blockNumber: limitOrder[i].blockNumber,
+                            isReduce: 0,
```
Comment (DesertEaglePosition) [inline contracts/protocol/libraries/position/PositionHouseFunction.sol:412]: add isReduce data in return when getListOrderPending for cancelLimitOrder

File: contracts/protocol/PositionHouse.sol [hunk 4]
```diff
@@ -341,15 +341,15 @@ contract PositionHouse is
     function cancelLimitOrder(
         IPositionManager _positionManager,
         uint64 _orderIdx,
-        bool _isReduce
+        uint8 _isReduce
```
Comment (DesertEaglePosition) [inline contracts/protocol/PositionHouse.sol:344]: change param type from bool to uint8 for client to call function

File: test/unit/PositionHouse_02.spec.ts [hunk 5]
```diff
@@ -267,12 +263,12 @@ describe("PositionHouse_02", () => {
         expect(positionData.quantity).eq(0);
     }
 
-    async function cancelLimitOrder(positionManagerAddress: string, trader: SignerWithAddress, orderId : string, pip : string, isReduce : boolean) {
+    async function cancelLimitOrder(positionManagerAddress: string, trader: SignerWithAddress, orderId : string, pip : string) {
```
Comment (DesertEaglePosition) [inline test/unit/PositionHouse_02.spec.ts:266]: update cancelLimitOrder in unit test

File: contracts/protocol/libraries/position/PositionHouseFunction.sol (hunk 2 above)
Comment (JustInPosition) [inline contracts/protocol/libraries/position/PositionHouseFunction.sol]:
```suggestion
        PositionLimitOrder.Data[] memory limitOrders,
```

File: contracts/protocol/libraries/position/PositionHouseFunction.sol [hunk 6]
```diff
@@ -378,39 +378,40 @@ library PositionHouseFunction {
     function getListOrderPending(
         address addressPositionManager,
         address _trader,
-        PositionLimitOrder.Data[] memory listLimitOrder,
+        PositionLimitOrder.Data[] memory limitOrder,
```
Comment (JustInPosition) [inline contracts/protocol/libraries/position/PositionHouseFunction.sol]:
```suggestion
        PositionLimitOrder.Data[] memory limitOrders,
```

**Pull request merged**